job/service.go

```diff
@@ -54,7 +55,19 @@ type JobRepoFactory interface {
 
 // ReplaySpecRepoFactory is used to manage replay spec objects from store
 type ReplaySpecRepoFactory interface {
-	New(jobSpec models.JobSpec) store.ReplaySpecRepository
+	New() store.ReplaySpecRepository
+}
+
+// ProjectRepoFactory is used to manage projects from store
+type ProjectRepoFactory interface {
+	New() store.ProjectRepository
+}
+
+type ReplayManager interface {
+	Init()
+	Replay(context.Context, models.ReplayRequest) (string, error)
+	GetReplay(uuid uuid.UUID) (models.ReplaySpec, error)
+	GetRunStatus(ctx context.Context, replaySpec models.ReplaySpec, jobSpec models.JobSpec) ([]models.JobStatus, error)
```

Comment:
I see replay spec already has job spec then why are we explicitly passing it again as an argument?

Reply:
updated to not use replaySpec and replaced with the necessary fields to avoid confusion